applications/GeoMechanicsApplication/custom_conditions/T_condition.cpp

```diff
@@ -0,0 +1,141 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_conditions/T_condition.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition() : Condition()
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+    : Condition(NewId, pGeometry)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId,
+                                        GeometryType::Pointer pGeometry,
+                                        PropertiesType::Pointer pProperties)
+    : Condition(NewId, pGeometry, pProperties)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::~TCondition() = default;
+
+template <unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TCondition<TDim, TNumNodes>::Create(IndexType NewId,
+                                                       NodesArrayType const& rThisNodes,
+                                                       PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(
+        new TCondition(NewId, GetGeometry().Create(rThisNodes), pProperties));
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::GetDofList(DofsVectorType& rConditionDofList,
+                                             const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    if (rConditionDofList.size() != TNumNodes) {
+        rConditionDofList.resize(TNumNodes);
+    }
+
+    const GeometryType& rGeom = GetGeometry();
+    for (unsigned int i = 0; i < TNumNodes; ++i) {
+        rConditionDofList[i] = rGeom[i].pGetDof(TEMPERATURE);
+    }
+
+    KRATOS_CATCH("")
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                                       VectorType& rRightHandSideVector,
+                                                       const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (rLeftHandSideMatrix.size1() != TNumNodes || rLeftHandSideMatrix.size2() != TNumNodes) {
+        rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+    }
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(TNumNodes, TNumNodes);
```

Comment:
My understanding is that the primary aim of the `noalias` syntax is to eliminate temporaries where the programmer sees this can be done safely. From the [ublas documentation](https://boostorg.github.io/ublas/) (see the Section named "Aliases"):

> Abstract formulas on vectors and matrices normally compose a couple of unary and binary operations. The conventional way of evaluating such a formula is first to evaluate every leaf operation of a composition into a temporary and next to evaluate the composite resulting in another temporary. This method is expensive in terms of time especially for small and space especially for large vectors and matrices. [...]

> One may get serious side effects using element wise evaluation on vectors or matrices. [...]

> Our solution for this problem _(i.e. where during the evaluation elements are overwritten that affect other evaluations)_ is to evaluate the right hand side of an assignment into a temporary and then to assign this temporary to the left hand side. [...] By using this syntax (i.e. `noalias`) a programmer can confirm, that the left and right hand sides of an assignment are independent, so that element wise evaluation and direct assignment to the target is safe.

The documentation doesn't mention that using `noalias` prevents resizing of vectors and matrices, as observed by @markelov208. I need to investigate that behavior in more detail first before making any statements about that. However, for this particular case, where we assign a zero matrix, I don't directly see any benefit in using `noalias`. Therefore, I would suggest to remove the `noalias` here, as well as the preceding resizing operation, since it is confusing and not strictly necessary. Do you agree @markelov208 and @rfaasse?

Reply:
I agree with removing noalias in that case

Reply:
I removed resize and noalias